frontend/src/metabase/lib/formatting.js

```diff
@@ -64,20 +78,54 @@ function formatMajorMinor(major, minor, options = {}) {
     }
 }
 
-export function formatTimeWithUnit(value, unit, options = {}) {
+export function formatTimeRangeWithUnit(value: Value, unit: DatetimeUnit, options: FormattingOptions = {}) {
     let m = parseTimestamp(value, unit);
     if (!m.isValid()) {
         return String(value);
     }
+
+    // Tooltips should show full month name, but condense "MMMM D, YYYY - MMMM D, YYYY" to "MMMM D - D, YYYY" etc
+    const monthFormat = options.type === "tooltip" ? "MMMM" : "MMM";
+    const condensed = options.type === "tooltip";
+    // use en dashes, for Maz
```

Comment:
Hmm it does add an additional DOM node per cell, that's the only reason I can think of it being less smooth. That seemed necessary to get both the contents aligned vertically, and the ellipsis behavior.

We don't need to use flex to align it vertically, it just seemed cleaner than picking magic padding sizes.